api/v1alpha1/virtualmachineimage_types.go

```diff
@@ -41,6 +41,26 @@ type VirtualMachineImageOSInfo struct {
 	Type string `json:"type,omitempty"`
 }
 
+// VirtualMachineImageInfo describes basic information related to the image, such as id and name. This information
+// can be used to identify an image in the content source.
+type VirtualMachineImageInfo struct {
+	//ImageID describes the ID of this VirtualMachineImage.
+	ImageID string `json:"imageId"`
+
+	// ImageName describes the display name of this VirtualMachineImage.
+	ImageName string `json:"imageName,omitempty"`
+}
+
+// VirtualMachineImageSourceInfo describes the uuid and display name related to the source of this image.
+type VirtualMachineImageSourceInfo struct {
```

Comment:
I think there ends up being a little stuttering with the `VirtualMachineImageSourceInfo` name, and then using `ImageSource` as the prefix for the members. Also, already in the top-level Spec we have a `ImageSourceType` so we've spread out the source info.

Also, seems this should be more like a `ContentSourceRef` because just the `ImageSourceID` isn't necessarily sufficient to identify the source. This`ImageSourceID` and the `Spec.ImageSourceType` tuple may be but how they are grouped is weird. 

A few options might be 
 - Just inline the fields here in the Spec along side the existing `ImageSourceID`
 - Deprecate `Spec.ImageSourceID` and move it here
 - Introduce a full Ref struct here to point to the content library. I think today we end up using the OwnerRefs to infer that.

@aruneshpa should chime in

Reply:
I agree with the stuttering here. `ImageSouurceInfo.ImageSource*` is not ideal. My preference would be to go with something like: `VirtualMachineImageSpec.Soruce.ID`. 

>  seems this should be more like a ContentSourceRef because just the ImageSourceID isn't necessarily sufficient to identify the source.

Yes. It will be complicated to identify a source just by looking at the ID (since we don't include a `type` here). I think you should consider including a `ContentProviderRef` here, since that has all the necessary info like the ID and the type. We can  modify the ContentProvider type to contain the display name as well in the `Status`.  And, the `VirtualMachineImage` to `ContentProvider` relation is expected and well defined.